charts/headlamp/templates/deployment.yaml

```diff
@@ -76,16 +98,27 @@ spec:
             {{- with .Values.config.pluginsDir}}
             - "-plugins-dir={{ . }}"
             {{- end }}
-            {{- if or .Values.config.oidc.clientID (not .Values.config.oidc.secret.create) }}
+            {{- if or (and .Values.config.oidc.clientID (not .Values.config.oidc.secret.create)) (and .Values.config.oidc.secret.create (ne .Values.config.oidc.clientID "")) }}
```

Comment:
Is it my impression or does this just translates to `{{- if ne .Values.config.oidc.clientID "" -}}`? What am I missing?
(same for the similar logic below)

Reply:
While it's true that both expressions check whether the `clientID` value is not empty, the original expression has additional conditions that may affect the outcome.

This expression contains two `and` conditions combined with an `or` condition:

- The first `and` condition:

Checks if `.Values.config.oidc.clientID` is truthy (i.e., not empty or false).
Checks if `.Values.config.oidc.secret.create` is falsy (i.e., evaluates to false or an empty value).

- The second and condition:

Checks if `.Values.config.oidc.secret.create` is truthy (i.e., not empty or false).
Checks if `.Values.config.oidc.clientID` is not equal to an empty string ("").

The `or` condition combines the two `and` conditions, meaning that the if block will execute if either of the `and` conditions evaluates to true.

The expression `{{- if ne .Values.config.oidc.clientID "" -}}` only checks whether `.Values.config.oidc.clientID` is not equal to an empty string (""). It does not consider the `secret.create` condition at all.